### DISCUSSION TITLE
dependency injection examples

### DISCUSSION THREAD

**Discussion by jordmantheman** [ai/nanostores#162] [General]
My first foray into nano-stores and I'm developing some common business logic that can be shared across a bunch of micro-frontends.

I can see my future self wanting to define some "stores" with micro-frontends as consumers and various shell/host apps as providers each providing their own custom business logic.

This is akin to dependency injection -- consumers (the react components) all code to an interface, while the shell app provides the implementation.

It **feels** like nanostores could fit into this paradigm easily, but I'm having a hard time imagining what this would look like using the existing building blocks.

Any thoughts from experienced users?

**Comment by ai** [comment]
Yes, the idea of Nano Stores is to move logic from React components to state managers.

I am thinking about an article explaining the idea with examples, but it will be ready only a few months later.

The general idea is to move logic into `computed()` stores and `onMount()`/`onSet()` events. If you give some example of the logic (for instance, React component), I will be able to show a store with the same logic.
